src/linux_serial_ccsds/linux_serial_ccsds.cpp

```diff
@@ -0,0 +1,226 @@
+        default:
+            *cflags |= B115200;
+            std::cerr << "Not supported baudrate value, defaulting to 115200\r\n";
+    }
+}
+
+inline void
+linux_serial_ccsds_private_data::driver_init_character_size(const Serial_CCSDS_Linux_Conf_T* const device, int* cflags)
+{
+    switch(device->bits) {
+        case 5:
+            *cflags |= CS5;
+            break;
+        case 6:
+            *cflags |= CS6;
+            break;
+        case 7:
+            *cflags |= CS7;
+            break;
+        case 8:
+            *cflags |= CS8;
+            break;
+        default:
+            *cflags |= CS8;
+            std::cerr << "Not supported character size, defaulting to 8 bits\r\n";
+    }
+}
+
+inline void
+linux_serial_ccsds_private_data::driver_init_parity(const Serial_CCSDS_Linux_Conf_T* const device, int* cflags)
+{
+    if(device->use_paritybit) {
+        *cflags |= PARENB;
+        switch(device->parity) {
+            case odd:
+                *cflags |= PARODD;
+                break;
+            case even:
+                *cflags &= ~PARODD;
+                break;
+            default:
+                *cflags &= ~PARENB;
+                std::cerr << "Not supported parity type deaulting to no parity";
+        }
+    } else {
+        *cflags &= ~PARENB;
+    }
+}
+
+void
+linux_serial_ccsds_private_data::driver_init(const SystemBus bus_id,
+                                             const SystemDevice device_id,
+                                             const Serial_CCSDS_Linux_Conf_T* const device_configuration,
+                                             const Serial_CCSDS_Linux_Conf_T* const remote_device_configuration)
+{
+    m_serial_device_bus_id = bus_id;
+    m_serial_device_id = device_id;
+    m_serial_device_configuration = device_configuration;
+    m_serial_remote_device_configuration = remote_device_configuration;
+
+    /// Open UART device
+    /**
+     * Access mode		O_RDWR - read write access mode
+     * Blocking mode	O_NDELAY - non blocking mode
+     * 					O_NOCTTY - pathname will refer to tty
+     */
+    serialFd = open(device_configuration->devname, O_RDWR | O_NOCTTY); //| O_NDELAY);
+    if(serialFd == -1) {
+        std::cerr << "Error while opening a file \n\r";
+        exit(EXIT_FAILURE);
+    }
+
+    /// Configure UART
+    struct termios options;
+
+    int cflags = 0;
+
+    driver_init_baudrate(device_configuration, &cflags);
+    driver_init_character_size(device_configuration, &cflags);
+    driver_init_parity(device_configuration, &cflags);
+
+    tcgetattr(serialFd, &options);
+    options.c_cflag = cflags | CLOCAL | CREAD;
+    options.c_iflag = IGNPAR;
+    options.c_oflag = 0;
+    options.c_lflag = 0;
+    tcflush(serialFd, TCIFLUSH);
+    tcsetattr(serialFd, TCSANOW, &options);
+
+    m_thread.start(&taste::LinuxSerialCcsdsPoll, this);
+}
+
+void
+linux_serial_ccsds_private_data::driver_poll()
+{
+    ssize_t length{ 0 };
+    Escaper_init_parser(&escaper);
```

Comment:
Should this be initialized multiple times?

Reply:
Based on linux_ip_socet I assume that escaper for parsing incoming messages should be initialized before polling loop (before receiving bytes from a remote device). @rphh, am I right?

Reply:
Yes, this should be initialized here